### PR TITLE
[GIE/Physical] Introduce new physical operators (in proto) and physical plan builder

### PR DESCRIPTION
**Pull request by BingqingLyu**
## What do these changes do?

1) Define new physical operators & plan in physical.proto;
2) Define new DataType for operators in type.proto;
3) Provide new physical plan builder.

## Related issue number

#2417

### CODE REVIEW

File: interactive_engine/executor/ir/proto/physical.proto [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  }
+  message Broadcast {}
+  oneof Strategy {
+    Shuffle to_another  = 1;
+    Broadcast to_others  = 2;
+  }
+}
+
+// A dummy node to delegate a source opr for multiple scan cases.
+message Dummy {}
```
Comment (longbinlai) [inline interactive_engine/executor/ir/proto/physical.proto:259]: Name it as `DummyScan`

File: interactive_engine/executor/ir/proto/physical.proto [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+syntax = "proto3";
+package physical;
+option java_package = "com.alibaba.graphscope.gaia.proto";
+option java_outer_classname = "GraphAlgebraPhysical";
+
+import "common.proto";
+import "expr.proto";
+import "schema.proto";
+import "type.proto";
+import "algebra.proto";
```
Comment (longbinlai) [inline interactive_engine/executor/ir/proto/physical.proto:26]: Suggest redefine those operators even they are the same as the logical ones.

File: interactive_engine/executor/ir/proto/physical.proto (hunk 1 above)
Comment (BingqingLyu) [reply interactive_engine/executor/ir/proto/physical.proto]: Done.

File: interactive_engine/executor/ir/proto/physical.proto [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+      Intersect intersect = 13;
+      Repartition repartition = 14;
+      RootScan root = 16;
+      // Saving the room for relational operators
+      GetV vertex = 30;
+      EdgeExpand edge = 31;
+      PathExpand path = 32;
+    }
+  }
+  message ColumnType {
```
Comment (longbinlai) [inline interactive_engine/executor/ir/proto/physical.proto:285]: 这个建议叫metadata。

Comment (BingqingLyu) [reply interactive_engine/executor/ir/proto/physical.proto]: Done

**Review by longbinlai — APPROVED** [review]

**Pull request merged**